js/main.js

```diff
@@ -80,23 +80,38 @@ function addEntry(event) {
 window.addEventListener('DOMContentLoaded', addEntry);
 
 var $body = document.querySelector('body');
-var $newModal = document.querySelector('.new');
-var $viewEntries = document.querySelector('.view');
-
-function switchView(event) {
-  if (event.target.matches('.entries-link')) {
-    $viewEntries.className = 'view';
-    $newModal.className = 'container hidden new';
-  } else if (event.target.matches('.new-button')) {
-    $viewEntries.className = 'view hidden';
-    $newModal.className = 'container new';
+var $view = document.querySelectorAll('.view');
+
+function switchView(name) {
+  data.view = name;
+  for (var i = 0; i < $view.length; i++) {
+    if (name === $view[i].getAttribute('data-view')) {
+      $view[i].className = 'view container';
+    } else {
+      $view[i].className = 'view container hidden';
+    }
   }
 }
 
-$body.addEventListener('click', switchView);
+function linkSwitch(event) {
+  if (event.target.matches('.link')) {
+    if (event.target.getAttribute('data-view') === 'entries') {
+      switchView('entries');
+    } else {
+      switchView('entry-form');
+    }
```

Comment:
You can just save the value of `event.target.getAttribute('data-view')` into a variable and pass that as an argument to `switchView`, which should avoid the need to have a conditional here